addons/coupon/models/coupon_program.py

```diff
@@ -144,3 +144,18 @@ def _get_valid_products(self, products):
             domain = ast.literal_eval(self.rule_products_domain)
             return products.filtered_domain(domain)
         return products
+
+    def _generate_coupons(self, partner_id):
```

Comment:
Why is it in coupon, and only used in pos_coupon? Should we use it in sale_coupon or put this one in pos_coupon?

Reply:
Maybe I was thinking that this method is generic enough that it can be put here. Indeed it is only used in pos_coupon. Would you recommend that I move this to pos_coupon?